File: app/src/main/java/com/example/feel_journal/data/remote/dto/EmotionEntryDtos.kt
package com.example.feel_journal.data.remote.dto

import com.google.gson.annotations.SerializedName

data class CreateEmotionEntryRequest(
    @SerializedName("date") val date: String,
    @SerializedName("note") val note: String?,
    @SerializedName("categoryId") val categoryId: Int,
    @SerializedName("emotionId") val emotionId: Int
)

data class UpdateEmotionEntryRequest(
    @SerializedName("date") val date: String,
    @SerializedName("note") val note: String?,
    @SerializedName("categoryId") val categoryId: Int,
    @SerializedName("emotionId") val emotionId: Int
)

data class EmotionEntryDto(
    @SerializedName("id") val id: Int,
    @SerializedName("date") val date: String,
    @SerializedName("note") val note: String?,
    @SerializedName("userId") val userId: Int,
    @SerializedName("categoryId") val categoryId: Int,
    @SerializedName("emotionId") val emotionId: Int,
    @SerializedName("category") val category: CategoryDto? = null,
    @SerializedName("emotion") val emotion: EmotionDto? = null,
    @SerializedName("createdAt") val createdAt: String,
    @SerializedName("updatedAt") val updatedAt: String
)

data class CategoryDto(
    @SerializedName("id") val id: Int,
    @SerializedName("name") val name: String,
    @SerializedName("type") val type: String
)

data class EmotionDto(
    @SerializedName("id") val id: Int,
    @SerializedName("name") val name: String,
    @SerializedName("icon") val icon: String
)

data class PagedResponse<T>(
    @SerializedName("items") val items: List<T>,
    @SerializedName("page") val page: Int,
    @SerializedName("pageSize") val pageSize: Int,
    @SerializedName("totalCount") val totalCount: Int,
    @SerializedName("totalPages") val totalPages: Int,
    @SerializedName("hasNext") val hasNext: Boolean,
    @SerializedName("hasPrev") val hasPrev: Boolean
)

File: server/DTOs/EmotionEntryDtos.cs
using System.ComponentModel.DataAnnotations;

namespace FeelJournal.DTOs;

public class EmotionEntryDto
{
    public int Id { get; set; }
    public string Date { get; set; } = string.Empty;
    public string? Note { get; set; }
    public int UserId { get; set; }
    public int CategoryId { get; set; }
    public int EmotionId { get; set; }
    public CategoryDto? Category { get; set; }
    public EmotionDto? Emotion { get; set; }
    public string CreatedAt { get; set; } = string.Empty;
    public string UpdatedAt { get; set; } = string.Empty;
}

public class CreateEmotionEntryRequest
{
    [Required]
    public string Date { get; set; } = string.Empty;

    public string? Note { get; set; }

    [Required]
    public int CategoryId { get; set; }

    [Required]
    public int EmotionId { get; set; }
}

public class UpdateEmotionEntryRequest
{
    [Required]
    public string Date { get; set; } = string.Empty;

    public string? Note { get; set; }

    [Required]
    public int CategoryId { get; set; }

    [Required]
    public int EmotionId { get; set; }
}

public class PagedResponse<T>
{
    public List<T> Items { get; set; } = new();
    public int Page { get; set; }
    public int PageSize { get; set; }
    public int TotalCount { get; set; }
    public int TotalPages => (int)Math.Ceiling(TotalCount / (double)PageSize);
    public bool HasNext => Page < TotalPages;
    public bool HasPrev => Page > 1;
}
